### PR TITLE
Fix minisearch not appear when disable suggestions search

### LINKED ISSUE #30584: mini search field head does not appear if suggestions and recommendations are disabled. Luma Theme and mobile navigation.
### Preconditions (*)

Magento 2.4-develop

1. does not work when suggestions are disabled
2. because form-mini.js is not loaded
3. imput depends on the click to appear and the logic is in form-mini.js

<?php if ($helper->isSuggestionsEnabled()): ?>
           data-mage-init='{"quickSearch":{data-mage-init='{"quickSearch":{
 
problem here
                                   ...
<?php endif; ?>	  

### Steps to reproduce (*)

1. Go to Admin -> Stores -> Configuration -> Catalog -> Catalog -> Catalog Search and set:

- **Enable Search Recommendations** to **No**
- **Enable Search Suggestions** to **No**

2. Click on the **Save Config** button;
3. Clean cache;
4. Using the standard luma theme, browse with mobile.
5. Click on the search button;

### Expected result (*)

 Appear input form search (form minisearch)

### Actual result (*)
:heavy_multiplication_x: Nothing happens

Peek 2020-10-22 12-37

---

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [S4] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by mrtuvn**
### Description (*)

Previously when owner shop disable suggestions search in backend config when user try search in mobile screen. Mini search will not active. User will not able to search thing

Use declarative syntax `data-mage-init` for improve performance instead script tag `x-magento-init`

### Preconditions (*)

Magento 2.4-develop
Php 7.4

### Fixed Issues (if relevant)

1. Fixes magento/magento2#30584
### Manual testing scenarios (*)

1. Go to admin
2. Catalog > Catalog Search 
3. Enable Search Suggestions Change to No
4. View storefront in real device mobile screen or resize browser screen
5. Click on search icon to see problem

### Expected result (*)

 Appear input form search (form minisearch)

### Actual result (*)
:heavy_multiplication_x: Nothing happens

Peek 2020-10-22 12-37
### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by rogyar** [comment]
@magento give me 2.4-develop instance

**Comment by rogyar** [comment]
@magento Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by rogyar** [comment]
Hi @mrtuvn, great job! Do you think we can cover this fix with auto-tests somehow?

**Comment by mrtuvn** [comment]
Sure let's cover it with some tests

**Comment by mrtuvn** [comment]
@rostyslav-hymon Can you let me know issue related with your change previously ? Why need to wrap quickSearch component with logic check suggestions search enable ?

**Comment by rostyslav-hymon** [comment]
> @rostyslav-hymon Can you let me know issue related with your change previously ? Why need to wrap quickSearch component initializing with php tag for check suggestions search enable ?
> [de9318d](https://github.com/magento/magento2/commit/de9318d7f41e3db30f03ef0bdabb1ae64d27f122)
> CC: @rogyar

Hi @mrtuvn . My change resolved the issue where disabling "Search Suggestions" did not have an effect on "Quick Search" behavior. It looks like your code solves this problem more correctly.

**Comment by mrtuvn** [comment]
@engcom-Charlie What test type should i start for such case ? MFTF

**Comment by engcom-Charlie** [comment]
@mrtuvn yes, via MFTF.

**Comment by mrtuvn** [comment]
Can you help with coverage test by mftf

**Comment by engcom-Charlie** [comment]
@mrtuvn you can set-up screen like here https://github.com/magento/magento2/blob/f472536749ce33aee355ba30eccf8227b0012dc3/app/code/Magento/Catalog/Test/Mftf/Test/StorefrontEnsureThatAccordionAnchorIsVisibleOnViewportOnceClickedTest.xml#L163
you can turn off search suggest for example like this
`<magentoCLI command="config:set catalog/search/search_suggestion_enabled 0" stepKey="disableSearchSuggestion"/>`

**Comment by mrtuvn** [comment]
@engcom-Charlie added mftf but can you check it again ? Let me know if you need update

**Comment by mrtuvn** [comment]
@magento run Static Tests, Unit Tests, Functional Tests CE, Functional Tests EE, Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/code/Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+        <before>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <!-- Disable search suggestion -->
+            <magentoCLI command="config:set catalog/search/search_suggestion_enabled 0" stepKey="disableSearchSuggestion"/>
+        </before>
+
+        <after>
+            <actionGroup ref="CliCacheFlushActionGroup" stepKey="flushCache">
```
Comment (rogyar) [inline app/code/Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml]: Please, do not flush the cache. In 99% cases, this action is not required but it slows down all other tests significantly. Please, clean the proper cache tags only

https://github.com/magento/magento2/blob/e9190a5bb2c4aede2a752e94a3e87dcf9647b604/app/code/Magento/Backend/Test/Mftf/ActionGroup/CliCacheCleanActionGroup.xml#L11

You will not slow down the testing process using this way. 

Thanks

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml (hunk 1 above)
Comment (mrtuvn) [reply app/code/Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml]: Updated my PR

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests EE

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8464](https://jira.corp.magento.com/browse/ENGCOM-8464) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed

**Manual testing scenario:**

1. Go to Admin -> Stores -> Configuration -> Catalog -> Catalog -> Catalog Search and set:
2. Enable Search Recommendations to No
3. Enable Search Suggestions to No
4. Click on the Save Config button;
5. Clean cache;
6. Using the standard luma or blank theme, browse with mobile.
7. Click on the search button;

**Before:** :heavy_multiplication_x:  Nothing happens

Peek 2020-11-16 15-27

**After:** :heavy_check_mark: Appear input form search (form minisearch)

_Luma_
Peek 2020-11-16 15-31

_Blank_
Peek 2020-11-16 15-37

**Review by engcom-Alfa — CHANGES_REQUESTED** [review]
Hi @mrtuvn .

Could you fix?

1. Move mftf test from `Section` to `Test` folder (`Magento/Search/Test/Mftf/Section/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml` -> `Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml` );

2. Change the name of the test from `StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabled.xml` to `StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml`);

3. Change the `title` and `description ` from `"Mini search field does not appear if suggestions was disabled"` to `"Mini search field appears if suggestions was disabled"` because the search appears with a fix

Thanks!

**Comment by mrtuvn** [comment]
@engcom-Alfa updated as your suggestions

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by engcom-Alfa** [comment]
Hi @rogyar .

Could you review the latest changes?
Thanks!

**Comment by engcom-Charlie** [comment]
Hi @mrtuvn, looks like your test `StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest` falls could you please take a look?
Thank you.

**Comment by mrtuvn** [comment]
Sure i will check that

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B, Functional Tests CE, Functional Tests EE, Integration Tests

**Comment by mrtuvn** [comment]
@magento run Functional Tests CE, Integration Tests

**Comment by mrtuvn** [comment]
@engcom-Alfa @engcom-Charlie any ideas why so much functional tests and integration broken

**Comment by mrtuvn** [comment]
@magento run all tests

File: app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            <severity value="CRITICAL"/>
+            <testCaseId value="MC-38652"/>
+            <group value="mtf_migrated"/>
+        </annotations>
+
+        <before>
+            <!-- Login as admin -->
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+            <!-- Disable search suggestion -->
+            <magentoCLI command="config:set catalog/search/search_suggestion_enabled 0" stepKey="disableSearchSuggestion"/>
```
Comment (rogyar) [inline app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml:25]: Hi @mrtuvn 
To persist the test isolation, we need to rollback all the changes (side-effects) introduced in the test. So, set the search suggestions back to "enabled" and logout from the admin panel. A common practice is to do it in the `<after>` section. 

Thanks.

**Review by rogyar — CHANGES_REQUESTED** [review]

File: app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml (hunk 2 above)
Comment (mrtuvn) [reply app/code/Magento/Search/Test/Mftf/Test/StorefrontVerifySearchFieldVisibilityWhenSuggestionDisabledTest.xml:25]: Updated as your suggest

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8464](https://jira.corp.magento.com/browse/ENGCOM-8464) has been created to process this Pull Request

**Comment by mrtuvn** [comment]
@magento run Integration Tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
rebased + squashed commits

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
@magento run all tests

**Comment by engcom-Charlie** [comment]
@magento run all tests

**Comment by mrtuvn** [comment]
Hi guys any ideas why Unit And Integration tests fails ? But all green when view reports

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Comment by mrtuvn** [comment]
@magento run Functional Tests B2B

**Review by rogyar — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @rogyar, thank you for the review.
[ENGCOM-8464](https://jira.corp.magento.com/browse/ENGCOM-8464) has been created to process this Pull Request

**Comment by engcom-Alfa** [comment]
:heavy_check_mark: QA Passed
The result is the same as in https://github.com/magento/magento2/pull/30634#issuecomment-728744111

**Pull request merged**